protocol/src/lib.rs

```diff
@@ -592,53 +592,55 @@ impl Message {
-            Message::UuidMismatch { .. } => None,
-            Message::Ruok { .. } => None,
-            Message::Imok { .. } => None,
-            Message::ExtentClose { .. } => None,
-            Message::ExtentReopen { .. } => None,
-            Message::ExtentFlush { .. } => None,
-            Message::ExtentRepair { .. } => None,
-            Message::RepairAckId { .. } => None,
-            Message::RegionInfoPlease { .. } => None,
-            Message::RegionInfo { .. } => None,
-            Message::ExtentVersionsPlease { .. } => None,
-            Message::ExtentVersions { .. } => None,
-            Message::LastFlush { .. } => None,
-            Message::LastFlushAck { .. } => None,
-            Message::Write { .. } => None,
-            Message::ExtentLiveClose { .. } => None,
-            Message::ExtentLiveFlushClose { .. } => None,
-            Message::ExtentLiveRepair { .. } => None,
-            Message::ExtentLiveReopen { .. } => None,
-            Message::ExtentLiveNoOp { .. } => None,
-            Message::Flush { .. } => None,
-            Message::ReadRequest { .. } => None,
-            Message::WriteUnwritten { .. } => None,
-            Message::Unknown(..) => None,
-
-            Message::ExtentError { error, .. } => Some(error),
-            Message::ErrorReport { error, .. } => Some(error),
+            Message::HereIAm { .. }
+            | Message::YesItsMe { .. }
+            | Message::VersionMismatch { .. }
+            | Message::ReadOnlyMismatch { .. }
+            | Message::EncryptedMismatch { .. }
+            | Message::PromoteToActive { .. }
+            | Message::YouAreNowActive { .. }
+            | Message::YouAreNoLongerActive { .. }
+            | Message::UuidMismatch { .. }
+            | Message::Ruok { .. }
+            | Message::Imok { .. }
+            | Message::ExtentClose { .. }
+            | Message::ExtentReopen { .. }
+            | Message::ExtentFlush { .. }
+            | Message::ExtentRepair { .. }
+            | Message::RepairAckId { .. }
+            | Message::RegionInfoPlease { .. }
+            | Message::RegionInfo { .. }
+            | Message::ExtentVersionsPlease { .. }
+            | Message::ExtentVersions { .. }
+            | Message::LastFlush { .. }
+            | Message::LastFlushAck { .. }
+            | Message::Write { .. }
+            | Message::ExtentLiveClose { .. }
+            | Message::ExtentLiveFlushClose { .. }
+            | Message::ExtentLiveRepair { .. }
+            | Message::ExtentLiveReopen { .. }
+            | Message::ExtentLiveNoOp { .. }
+            | Message::Flush { .. }
+            | Message::ReadRequest { .. }
+            | Message::WriteUnwritten { .. }
+            | Message::Unknown(..) => None,
+
+            Message::ExtentError { error, .. }
+            | Message::ErrorReport { error, .. } => Some(error),
 
             Message::ExtentLiveCloseAck { result, .. } => result.as_ref().err(),
```

Comment:
I don't know if this makes it any simpler, but I also don't really care one way or the other.  I do think it's all part of your master plan to touch every line in Crucible :)

However, you should be able to put `Message::ExtentLiveCloseAck` with the rest just below, right?

Reply:
> I don't know if this makes it any simpler, but I also don't really care one way or the other. I do think it's all part of your master plan to touch every line in Crucible :)

🤫 

> However, you should be able to put `Message::ExtentLiveCloseAck` with the rest just below, right?

They're actually different types `Result<(), CrucibleError>` versus `Result<(u64, u64, bool), CrucibleError>`, so you can't match on them simultaneously (even though we're only using the error).